src/pages/AddClientPage.tsx

```diff
@@ -0,0 +1,136 @@
+// import { useRef } from "react";
+import { useAddClient } from "../api/mutations";
+import { useFormik } from "formik";
+import { clientValidationSchema } from "../validators/validators";
+import TextField from "@mui/material/TextField";
+import { useNavigate } from "react-router-dom";
+import { ROUTES } from "../routes";
+
+//* DONE: podpiac formikiem + walidacja + usuwanie i edycja Klientów
+type FormValues = {
+  imgSrc: string;
+  name: string;
+  surname: string;
+  street: string;
+  postCode: string;
+  town: string;
+  region: string;
+  phoneNumber: string;
+};
+
+export const AddClientPage = () => {
+  const navigate = useNavigate();
+
+  const { mutate } = useAddClient();
+  const formik = useFormik<FormValues>({
+    initialValues: {
+      imgSrc: "",
+      name: "",
+      surname: "",
+      street: "",
+      postCode: "",
+      town: "",
+      region: "",
+      phoneNumber: "",
+    },
+    onSubmit: (values: FormValues) => {
+      mutate(values);
+      navigate(ROUTES.clients);
```

Comment:
w mutate po onSuccess powinno być przekierowanie, bo teraz przed zakończeniem mutacji możesz zrobić przekierowanie. Możesz albo spróbować przekazać onSuccess jako callback to useAddClient albo spróbować zrobić mutację przez mutateAsync